src/NServiceBus.Gateway/IGatewayDeduplicationPersistence.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace NServiceBus
+{
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using Extensibility;
+
+    /// <summary>
+    /// Defines the api to provide storage for the gateway deduplication mechanism.
+    /// </summary>
+    public interface IGatewayDeduplicationStorage
+    {
+        /// <summary>
+        /// Returns whether the storage can enlist in a distributed transaction via ambient <see cref="TransactionScope"/> to support exactly-once dispatching.
+        /// </summary>
+        bool SupportsDistributedTransactions { get; }
```

Comment:
Why is this needed? What logic would be different knowing the persister supports enlisting? In both cases, both methods would still be invoked.

Reply:
we discussed this on a call yesterday and concluded that we probably need this for now for correct handling of transaction scopes (e.g. ensuring that there is also no transaction scope when the persistence doesn't support it)